### PR TITLE
Deprecate View#nearestChildOf

### LINKED ISSUE #9641: nearestChildOf returns a parent view instead of a child
https://github.com/emberjs/ember.js/blob/v1.8.1/packages/ember-views/lib/views/view.js#L971-L986

(I can write a test for this)

### PR DESCRIPTION
**Pull request by tim-evans**
Judging from the code in nearestChildOf, it never worked. This formally deprecates this function to prevent confusion (fixes #9641).

### CODE REVIEW

**Comment by tim-evans** [comment]
@mixonic This should be better!

File: packages/ember-views/lib/views/view.js [hunk 1]
```diff
@@ -1022,6 +1022,8 @@ var View = CoreView.extend({
     @return Ember.View
   */
   nearestChildOf: function(klass) {
+    Ember.deprecate("nearestChildOf has been deprecated.\nFor existing behavior, use nearestOfType.");
```
Comment (mixonic) [inline packages/ember-views/lib/views/view.js]: I suggest `"nearestChildOf has been deprecated in favor of nearestOfType."`

**Comment by mixonic** [comment]
@tim-evans thanks for this and the test. Really important to get this cruft flagged so we can drop it in 2.0

File: packages/ember-views/lib/views/view.js (hunk 1 above)
Comment (tim-evans) [inline packages/ember-views/lib/views/view.js]: Technically, it's not in favor. I'd like to make it clear to people who try to call this (like myself) that this doesn't do what is expected while marking a clear deprecation path for those who are currently using this.

Comment (mixonic) [inline packages/ember-views/lib/views/view.js]: I'm open to an alternative. `For existing behavior` is pretty meaningless though. `For similar behavior` doesn't seem worth mentioning. Can you iterate a bit more?

Comment (tim-evans) [inline packages/ember-views/lib/views/view.js]: Eh. The existing behavior doesn't work. At all. So I might just change this to: `"nearestChildOf has been deprecated"`

**Comment by tim-evans** [comment]
@mixonic no problemo. This one bit me when I tried to use it (and it didn't work as expected / didn't work)

**Pull request merged**